items.js

```diff
@@ -146,4 +146,143 @@ class Mushroom {
             ctx.strokeRect(this.BB.x - this.game.camera.x, this.BB.y, this.BB.width, this.BB.height);
         }
     };
-};
\ No newline at end of file
+};
+
+class Flower {
+    constructor(game, x, y, brick) {
+        Object.assign(this, { game, x, y, brick });
+        this.velocity = { x: 0, y: -PARAMS.BLOCKWIDTH };
+        this.emerging = true;
+        this.animation = new Animator(ASSET_MANAGER.getAsset("./sprites/items.png"), 4, 64, 16, 16, 4, 0.15, 14, false, true);
+        this.updateBB();
+    };
+
+    updateBB() {
+        this.BB = new BoundingBox(this.x, this.y, PARAMS.BLOCKWIDTH, PARAMS.BLOCKWIDTH);
+    };
+
+    update() {
+        if (this.emerging) {
+            this.y += this.game.clockTick * this.velocity.y * PARAMS.SCALE;
+
+            if (this.y < this.BB.top - PARAMS.BLOCKWIDTH) {
+                this.emerging = false;
+                this.velocity.x = PARAMS.BLOCKWIDTH;
+                this.velocity.y = 0;
+            }
+        } else {
+            this.updateBB();
+        }
+    };
+
+    drawMinimap(ctx, mmX, mmY) {
+    }
+
+   draw(ctx) {
+        this.animation.drawFrame(this.game.clockTick, ctx, this.x - this.game.camera.x, this.y, PARAMS.SCALE);
+        if (this.emerging) this.brick.draw(ctx);
+        if (PARAMS.DEBUG) {
+            ctx.strokeStyle = 'Red';
+            ctx.strokeRect(this.BB.x - this.game.camera.x, this.BB.y, this.BB.width, this.BB.height);
+        }
+    };
+};
+
+class Fireball {
```

Comment:
this class name does NOT conflict with the "fire" class that seoungdeok added for their "firebar" implementation, a enemy that makes fireballs float/rotate around a brick, figured I'd check and alert the two of you.